psydac/api/ast/evaluation.py

```diff
@@ -860,7 +878,7 @@ def _initialize(self):
         weights_elements = []
         if self.is_rational_mapping:
             # TODO check if 'w' exist already
-            weights_pts = ScalarField(self.space, name='w')
+            weights_pts = element_of(self.space, name='w')
```

Comment:
should we keep the name w hard coded here? or check if there's an atom of the same name, if it is the case, then we create a random string?

Reply:
See issue #55